### PR TITLE
race-control bugfix

### PR DESCRIPTION
**Pull request by Jakllp**
- Fixed a bug where a cart with race-controls could not be thrown into the air (e.g. by MarioKart-shells)

- Internal bump to 22.1

### CODE REVIEW

File: src/main/java/com/useful/ucars/uCarsListener.java [hunk 1]
```diff
@@ -611,8 +611,10 @@ public void tickCalcsAndLegacy(VehicleUpdateEvent event) {
 					vel.setY(5);
 				}
 				UEntityMeta.removeMetadata(car, "car.jumpFull");
+			} else if(UEntityMeta.hasMetadata(car, "car.frozen") || car.hasMetadata("car.frozen") && car.hasMetadata("kart.immune")) {
```
Comment (storm345) [inline src/main/java/com/useful/ucars/uCarsListener.java:614]: Is this needed? Surely if a car is frozen it should set the y velocity to 0 still, not leave it untouched? (Or it wouldn't be frozen?).  And if it is needed please add parenthesis for clarity

File: src/main/java/com/useful/ucars/uCarsListener.java [hunk 2]
```diff
@@ -879,7 +881,7 @@ public void onUcarUpdate(ucarUpdateEvent event) {
 			}
 		}
 		
-		Vector travel = event.getTravelVector(); // Travel Vector,
+		Vector travel = event.getTravelVector().setY(event.getVehicle().getVelocity().getY()); // Travel Vector,
```
Comment (storm345) [inline src/main/java/com/useful/ucars/uCarsListener.java:884]: Hmm, are you sure that this doesn't break things with block climbing and other situations where the car velocity is modified un-naturally? This also overwrites the y value calculated in "Motion Manager" (Yes this plugin's code really is a mess..). I suspect that with this conservation of y velocity after a block-climb or similar which perturbs the velocity momentarily to a large value the car may want to shoot up into the air. (There was an old bug that used to be that if you block-climbed at the same time as stopping pressing the accelerate button the car would shoot upwards as the y velocity would be conserved. Sometimes quite a few blocks into the air...)

**Review by storm345 — COMMENTED** [review]
See the review comments, I'm concerned that this change may make cars fly skywards when they shouldn't (since y velocity would now be conserved in some cases).

File: src/main/java/com/useful/ucars/uCarsListener.java (hunk 2 above)
Comment (Jakllp) [reply src/main/java/com/useful/ucars/uCarsListener.java:884]: That's interesting, I'll try to reproduce that.
The problem I was trying to address with these changes was that cars could not be thrown upwards (more than one block) by stuff like bombs in the MarioKart-Plugin after the last changes.

I'll look into it

Comment (storm345) [reply src/main/java/com/useful/ucars/uCarsListener.java]: I think there is an existing way to denote the car is being flung, but see what you can find I guess. Maybe not car.penalty but some other name that denotes the y velocity is conserved instead as it's more meaningful :P

Comment (Jakllp) [reply src/main/java/com/useful/ucars/uCarsListener.java]: What about "car.decoupleY"?

Comment (storm345) [reply src/main/java/com/useful/ucars/uCarsListener.java]: or car.inertialYAxis? or something like that?

**Comment by Jakllp** [comment]
car.inertialYAxis it is

**Pull request merged**